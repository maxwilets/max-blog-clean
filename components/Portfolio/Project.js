import { OpenLink } from "../../icons/OpenLink";
import { PortfolioStyles } from "./style";

export default function Project({ projectData }) {
  return (
    <PortfolioStyles>
      <h1>testing</h1>
      {/* <section className="sideBySide">
        <section className="imageContainer">
          <img
            src={projectData.photo.image.publicUrlTransformed}
            alt={projectData.photo.altText}
          />
        </section>

        <section className="contentContainer">
          <h3>{projectData.title}</h3>
          <p>{projectData.p1}</p>
          {projectData.p2 ? <p>{projectData.p2}</p> : null}
          {projectData.p3 ? <p>{projectData.p3}</p> : null}
          {projectData.repo ? (
            <a
              className="btn"
              target="_blank"
              href={projectData.repo}
              rel="noreferrer"
            >
              Repository
              <OpenLink />
            </a>
          ) : null}
          {projectData.liveSite ? (
            <a
              className="btn"
              target="_blank"
              href={projectData.liveSite}
              rel="noreferrer"
            >
              Repository
              <OpenLink />
            </a>
          ) : null}
        </section>
      </section>
      <h3>Technologies Used</h3>
      <section className="btnContainer">
        {projectData.technology.map((techButton) => (
          <a
            href={techButton.typeLink}
            key={techButton.id}
            target="_blank"
            className="btn"
            rel="noreferrer"
          >
            {techButton.name}
            <OpenLink />
          </a>
        ))}
      </section> */}
    </PortfolioStyles>
  );
}
