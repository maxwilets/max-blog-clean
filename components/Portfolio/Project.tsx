/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import PropTypes from 'prop-types';
import { useState } from 'react';
import { OpenLink } from '../../icons/OpenLink';
import { PortfolioStyles } from './style';
import { Minus } from '../../icons/Minus';
import { PlusIcon } from '../../icons/PlusIcon';

type PhotoProps = {
  altText: string;
  image: { publicUrlTransformed: string };
};

type TechnologyProps = {
  id: string;
  name: string;
  typeLink: string;
};

type PropjectProps = {
  id: string;
  title: string;
  liveSite: string;
  p1: string;
  p2: string;
  p3: string;
  repo: string;
  photo: PhotoProps;
  technology: TechnologyProps[];
};
export default function Project({
  projectData,
}: {
  projectData: PropjectProps;
}): any {
  const [toggleState, updateToggleState] = useState('plus');
  function toggle() {
    return toggleState === 'plus'
      ? updateToggleState('minus')
      : updateToggleState('plus');
  }
  return (
    <PortfolioStyles>
      <section className="sideBySide">
        <section className="imageContainer">
          <a target="_blank" href={projectData.liveSite} rel="noreferrer">
            <img
              src={projectData.photo.image.publicUrlTransformed}
              alt={projectData.photo.altText}
            />
          </a>
        </section>

        <section className="contentContainer">
          <h3>{projectData.title}</h3>
          <p>{projectData.p1}</p>
          {projectData.p2 ? <p>{projectData.p2}</p> : null}
          {projectData.p3 ? <p>{projectData.p3}</p> : null}
          {projectData.repo ? (
            <a
              className="btn blue"
              target="_blank"
              href={projectData.repo}
              rel="noreferrer"
            >
              Repository
              <OpenLink />
            </a>
          ) : null}
          {projectData.liveSite ? (
            <a
              className="btn blue"
              target="_blank"
              href={projectData.liveSite}
              rel="noreferrer"
            >
              Live Site
              <OpenLink />
            </a>
          ) : null}
        </section>
      </section>
      {projectData.technology.length > 0 ? (
        <>
          <a className="noDecoration" onClick={toggle}>
            <h3 className="technlogyHeader">
              <span className="toggleIcon">
                {toggleState === 'plus' ? <PlusIcon /> : <Minus />}
              </span>
              Technologies Used
            </h3>
          </a>
          <section
            className={
              toggleState === 'plus'
                ? 'btn-container hide'
                : 'btn-container show'
            }
          >
            {projectData.technology.map((techButton) => (
              <a
                href={techButton.typeLink}
                key={techButton.id}
                target="_blank"
                className="btn"
                rel="noreferrer"
              >
                {techButton.name}
                <OpenLink />
              </a>
            ))}
          </section>
        </>
      ) : null}
    </PortfolioStyles>
  );
}

Project.propTypes = {
  projectData: PropTypes.object,
};
