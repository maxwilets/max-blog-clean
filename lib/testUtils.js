import casual from 'casual';
import { PAGINATION_QUERY } from '../components/Pagination';

// seed it so we get consistent results
casual.seed(777);

const fakeItem = () => ({
  // __typename: 'Item',
  id: 'abc123',
  price: 5000,
  user: null,
  photo: {
    id: 'abc123',
    altText: 'dogs are best',
    image: {
      publicUrlTransformed: 'dog.jpg',
    },
  },
  name: 'dogs are best',
  description: 'dogs',
});

const fakeUser = (overrides) => ({
  __typename: 'User',
  id: '4234',
  name: casual.name,
  email: casual.email,
  permissions: ['ADMIN'],
  orders: [],
  cart: [],
  ...overrides,
});

const fakeOrderItem = () => ({
  __typename: 'OrderItem',
  id: casual.uuid,
  image: {
    image: `${casual.word}.jpg`,
  },
  name: casual.words(),
  price: 4234,
  quantity: 1,
  description: casual.words(),
});

const fakeOrder = () => ({
  __typename: 'Order',
  id: 'ord123',
  charge: 'ch_123',
  total: 40000,
  items: [fakeOrderItem(), fakeOrderItem()],
  createdAt: '2022-12-11T20:16:13.797Z',
  user: fakeUser(),
});

const fakeProject = () => ({
  __typename: 'Project',
  id: 'proj123',
  title: 'test project',
  p1: 'here is a testing project testing UI',
  p2: 'here is the second paragraph testing the UI for Project',
  repo: 'https://github.com/maxwilets',
  liveSite: 'https://maxwilets.io',
  photo: {
    altText: 'mountians',
    image: {
      publicUrlTransformed:
        'https://res.cloudinary.com/dsna4d0gs/image/upload/v1640028217/samples/landscapes/nature-mountains.jpg',
    },
  },
  technology: [
    {
      id: 'technology123',
      name: 'javascript',
      typeLink: 'https://javascript.com',
    },
    {
      id: 'technology1234',
      name: 'typescript',
      typeLink: 'https;//typescript.com',
    },
  ],
});

const fakeCartItem = (overrides) => ({
  __typename: 'CartItem',
  id: 'omg123',
  quantity: 3,
  product: fakeItem(),
  user: fakeUser(),
  ...overrides,
});

// Fake LocalStorage
class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

function makePaginationMocksFor(length) {
  return [
    {
      request: { query: PAGINATION_QUERY },
      result: {
        data: {
          _allProductsMeta: {
            count: length,
          },
          itemsConnection: {
            __typename: 'aggregate',
            aggregate: {
              count: length,
              __typename: 'count',
            },
          },
        },
      },
    },
  ];
}

export {
  makePaginationMocksFor,
  LocalStorageMock,
  fakeItem,
  fakeUser,
  fakeCartItem,
  fakeOrder,
  fakeOrderItem,
  fakeProject,
};
