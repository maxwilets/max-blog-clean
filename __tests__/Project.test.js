import { render, screen } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import Project from '../components/Portfolio/Project';
import { fakeProject } from '../lib/testUtils';

const project = fakeProject();

describe('<Project/>', () => {
  it('renders out the description title technologies and image', () => {
    const { container, debug } = render(
      <MockedProvider>
        <Project projectData={project} />
      </MockedProvider>
    );
    const description = screen.getByText(project.p1);
    debug(description);
    expect(description).toBeInTheDocument();
    const link = container.querySelector('a');
    debug(link);
    expect(link).toHaveAttribute('href', 'https://github.com/maxwilets');
    expect(link).toHaveAttribute('target', '_blank');
    expect(link).toHaveTextContent('Repository');
  });

  it('Renders and matches the snapshot', () => {
    const { container, debug } = render(
      <MockedProvider>
        <Project projectData={project} />
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the image properly', () => {
    const { container, debug } = render(
      <MockedProvider>
        <Project projectData={project} />
      </MockedProvider>
    );
    // grab the image
    const img = screen.getByAltText(project.photo.altText);
    expect(img).toBeInTheDocument();
    debug(img);
  });
});
