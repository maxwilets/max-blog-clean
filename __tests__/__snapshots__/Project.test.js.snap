// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Project/> Renders and matches the snapshot 1`] = `
<div>
  <div
    class="style__PortfolioStyles-sc-1mnrp3x-0 hHsCNL"
  >
    <section
      class="sideBySide"
    >
      <section
        class="imageContainer"
      >
        <img
          alt="mountians"
          src="https://res.cloudinary.com/dsna4d0gs/image/upload/v1640028217/samples/landscapes/nature-mountains.jpg"
        />
      </section>
      <section
        class="contentContainer"
      >
        <h3>
          test project
        </h3>
        <p>
          here is a testing project testing UI
        </p>
        <p>
          here is the second paragraph testing the UI for Project
        </p>
        <a
          class="btn"
          href="https://github.com/maxwilets"
          rel="noreferrer"
          target="_blank"
        >
          Repository
          <svg
            fill="#fff"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.5 2.25a.75.75 0 01.75-.75h5.5a.75.75 0 01.75.75v5.5a.75.75 0 01-1.5 0V4.06l-6.22 6.22a.75.75 0 11-1.06-1.06L19.94 3h-3.69a.75.75 0 01-.75-.75z"
            />
            <path
              d="M2.5 4.25c0-.966.784-1.75 1.75-1.75h8.5a.75.75 0 010 1.5h-8.5a.25.25 0 00-.25.25v15.5c0 .138.112.25.25.25h15.5a.25.25 0 00.25-.25v-8.5a.75.75 0 011.5 0v8.5a1.75 1.75 0 01-1.75 1.75H4.25a1.75 1.75 0 01-1.75-1.75V4.25z"
            />
          </svg>
        </a>
        <a
          class="btn"
          href="https://maxwilets.io"
          rel="noreferrer"
          target="_blank"
        >
          Live Site
          <svg
            fill="#fff"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.5 2.25a.75.75 0 01.75-.75h5.5a.75.75 0 01.75.75v5.5a.75.75 0 01-1.5 0V4.06l-6.22 6.22a.75.75 0 11-1.06-1.06L19.94 3h-3.69a.75.75 0 01-.75-.75z"
            />
            <path
              d="M2.5 4.25c0-.966.784-1.75 1.75-1.75h8.5a.75.75 0 010 1.5h-8.5a.25.25 0 00-.25.25v15.5c0 .138.112.25.25.25h15.5a.25.25 0 00.25-.25v-8.5a.75.75 0 011.5 0v8.5a1.75 1.75 0 01-1.75 1.75H4.25a1.75 1.75 0 01-1.75-1.75V4.25z"
            />
          </svg>
        </a>
      </section>
    </section>
    <h3>
      Technologies Used
    </h3>
    <section
      class="btnContainer"
    >
      <a
        class="btn"
        href="https://javascript.com"
        rel="noreferrer"
        target="_blank"
      >
        javascript
        <svg
          fill="#fff"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.5 2.25a.75.75 0 01.75-.75h5.5a.75.75 0 01.75.75v5.5a.75.75 0 01-1.5 0V4.06l-6.22 6.22a.75.75 0 11-1.06-1.06L19.94 3h-3.69a.75.75 0 01-.75-.75z"
          />
          <path
            d="M2.5 4.25c0-.966.784-1.75 1.75-1.75h8.5a.75.75 0 010 1.5h-8.5a.25.25 0 00-.25.25v15.5c0 .138.112.25.25.25h15.5a.25.25 0 00.25-.25v-8.5a.75.75 0 011.5 0v8.5a1.75 1.75 0 01-1.75 1.75H4.25a1.75 1.75 0 01-1.75-1.75V4.25z"
          />
        </svg>
      </a>
      <a
        class="btn"
        href="https;//typescript.com"
        rel="noreferrer"
        target="_blank"
      >
        typescript
        <svg
          fill="#fff"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.5 2.25a.75.75 0 01.75-.75h5.5a.75.75 0 01.75.75v5.5a.75.75 0 01-1.5 0V4.06l-6.22 6.22a.75.75 0 11-1.06-1.06L19.94 3h-3.69a.75.75 0 01-.75-.75z"
          />
          <path
            d="M2.5 4.25c0-.966.784-1.75 1.75-1.75h8.5a.75.75 0 010 1.5h-8.5a.25.25 0 00-.25.25v15.5c0 .138.112.25.25.25h15.5a.25.25 0 00.25-.25v-8.5a.75.75 0 011.5 0v8.5a1.75 1.75 0 01-1.75 1.75H4.25a1.75 1.75 0 01-1.75-1.75V4.25z"
          />
        </svg>
      </a>
    </section>
  </div>
</div>
`;
